docs/core-concepts/networking.md

```diff
@@ -0,0 +1,177 @@
+
+## Publishing Services
+
+Accessing any Service externally requires the Service's ports to be explicitly published, using a list of published TCP and UDP ports, configured in the `kontena.yml`.
+These are deployed as published container ports on the Docker default bridge network, using DNAT on the host machine to provide external access to the published services.
+
+The host Node's public (and private) addresses can be used to access those services published by Service Containers running on that Node.
+Only one Service can publish any given port on a given Node.
+The ports used for infrastructure services on the host Nodes cannot be used to publish other services.
+
+### Kontena Load Balancer
+
+The [Kontena Load Balancer](https://kontena.io/docs/using-kontena/loadbalancer) can also be used to publish services, providing TCP and HTTP load-balancing of multiple Service Container backends with dynamic configuration as services are started, scaled and stopped.
+The public network address of any Host Node running the load balancer service can be used by external clients to connect to the load-balanced service containers.
+The Kontena Load Balancer should be deployed to nodes having known public network addresses, using either `kontena.yml` `affinity` conditions, or using `deploy: strategy: daemon` to deploy the LB on all nodes.
+The public network address of the nodes running the LB service can then be configured within any external services used for request routing, such as a DNS server.
+
+Kontena Services can be linked to any [Kontena Load Balancer](https://kontena.io/docs/using-kontena/loadbalancer) service (services using the `kontena/lb` image) within the same Grid.
+The Kontena Master will generate `io.kontena.load_balancer.*` labels for each such linked Service Container.
+The Kontena Agent on each Host Node uses the Docker Events API to register any running containers having such labels into the Grid's etcd database.
+The etcd database is used by the `kontena/lb` service for dynamic load balancer configuration.
+
+See the [Kontena Load Balancer](https://kontena.io/docs/using-kontena/loadbalancer) documentation for usage documentation.
+The [implementation](https://github.com/kontena/kontena-loadbalancer) of the Kontena Load Balancer is based on [HAProxy](http://www.haproxy.org/), using [confd](https://github.com/kelseyhightower/confd) for dynamic configuration.
+
+## DNS (`kontena.local`)
+
+Each Kontena Grid uses Weave DNS for dynamic service discovery of other service containers within the same grid.
+Each Service Container is configured with the `kontena.local` search domain, using the local node's `docker0` IP as the DNS resolver.
+Within the internal `kontena.local` DNS namespace, each Service Container is registered for both the per-container and per-service names under both the `kontena.local` and `$grid.kontena.local` names.
+
+The `kontena.local` names are deprecated and the `$grid.kontena.local` names should be used instead.
```

Comment:
Clarified this to recommend the use of the short DNS names or the fully-qualified names in preference to the older `.kontena.local` names.